packages/askui-nodejs/src/utils/proxy/proxy-utils.ts

```diff
@@ -0,0 +1,111 @@
+import fs from 'fs';
+import http from 'http';
+import https from 'https';
+import { join } from 'path';
+import dns, { LookupOneOptions } from 'dns';
+
+const proxy = require('proxy');
+
+export const PROXY_HOSTNAME = 'proxy.proxy-unit-test-askui.com';
+export const SERVER_HOSTNAME = 'server.proxy-unit-test-askui.com';
+
+/** *
+ * This file is oriented by the integration tests of hpagent: https://github.com/delvedor/hpagent/blob/main/test/utils.js
+ */
+export function addBasicAuthentication(httpProxy: http.Server): http.Server {
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any, no-param-reassign
+  (httpProxy as any).authenticate = (req: any, fn: any) => {
+    const header = req.headers['proxy-authorization'] || ''; // get the auth header
+    const token = header.split(/\s+/).pop() || ''; // and the encoded auth token
+    const auth = Buffer.from(token, 'base64').toString(); // convert from base64
+    const parts = auth.split(/:/); // split on colon
+    const username = parts.shift(); // username is first
+    const password = parts.join(':');
+
+    if (!(username === 'username' && password === 'password')) {
+      fn(null, false);
+      return;
+    }
+    fn(null, true);
```

Comment:
What is going on here? What is `fn` actually? I think some renaming, extracting functions and explicit typing could already help my understanding.

I don't get the whole `authenticate` function as we are not doing anything with the username and password extracted other than comparing them to some magic values.

Reply:
This is integration code to check how hpagent is working together with a proxy. 

Here we check `basic auth`  for the proxy.

Reply:
Put it into code ;)